l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -796,17 +804,41 @@ def exportFatturaPA(self):
             'type': 'ir.actions.act_window',
             }
 
-    def generate_attach_report(self, inv):
+    def generate_attach_report(self, invoices):
+        # Generate the report
+        if not self.report_print_menu:
+            return
         action_report_model, action_report_id = (
             self.report_print_menu.value.split(',')[0],
             int(self.report_print_menu.value.split(',')[1]))
         action_report = self.env[action_report_model] \
             .browse(action_report_id)
         report_model = self.env['report']
-        report_model.get_pdf(inv.ids, action_report.report_name)
-        attachment = report_model._attachment_stored(
-            inv, action_report)[inv.id]
-        inv.write({
-            'fatturapa_doc_attachments': [(0, 0, {
-                'ir_attachment_id': attachment.id})]
-        })
+
+        # Manage the attachment in the selected mode
+        if self.attach_mode in ('mail', 'both'):
+            for invoice in invoices:
+                composer_action = invoice.action_invoice_sent()
+                composer_model_name = composer_action['res_model']
+                composer_context = composer_action['context']
+                # Override composition mode so we can send silently
+                composer_context.update(composition_mode='mass_mail')
+                composer = self.env[composer_model_name] \
+                    .with_context(composer_context) \
+                    .create({})
+                composer.onchange_template_id_wrapper()
+                composer.send_mail()
+
+        if self.attach_mode in ('xml', 'both'):
+            # Generate the PDFs,
+            # they are automatically attached to the invoices
```

Comment:
@SimoRubi dipende: l'utente potrebbe aver disabilitato il salvataggio del PDF come allegato della fattura. Inoltre, se elimini l'XML e rilanci la procedura, il PDF verrebbe allegato 2 volte.

Divresti fare ciò che avviene quando lanci la stampa di una fattura: se il c'è il PDF allegato, usi quello, altrimenti lo generi (`get_pdf`)

Reply:
Grazie della review!
Se nel report disabiliti il 'ricarica da allegato' e poi nella fattura premi due volte il pulsante 'stampa' hai due allegati.
Lo stesso succede se elimini il file XML e lo ricrei quindi il comportamento è lo stesso.
Forse però non ho capito qual'è il problema che dici, mi diresti gli step per riprodurlo?

Ho anche guardato in [account.invoice.invoice_print](https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_invoice.py#L389) ma non si fa tanti problemi di controllare l'allegato

Reply:
> Se nel report disabiliti il 'ricarica da allegato' e poi nella fattura premi due volte il pulsante 'stampa' hai due allegati

Infatti in quel caso dovresti anche svuotare il campo `attachment`.
Però ho controllato meglio e forse è già gestito da `get_pdf`? Non mi è chiarissimo